File: 코드샘플/메타토이드래곤즈/유니티/System/Data/Datas/ElementData.cs
using Newtonsoft.Json.Linq;

namespace SandboxNetwork
{
    public class ElementRateData : TableData<DBElement_rate>
    {
        static private ElementTable table = null;
        static public ElementRateData Get(string key)
        {
            if (table == null)
                table = TableManager.GetTable<ElementTable>();

            return table.Get(key);
        }
        static public ElementRateData Get(eElementType type)
        {
            if (table == null)
                table = TableManager.GetTable<ElementTable>();

            return table.Get(type);
        }
        public string KEY => Data.UNIQUE_KEY;
        public string A_ELEMENT => Data.A_ELEMENT;
        public int T_FIRE => Data.T_FIRE;
        public int T_WATER => Data.T_WATER;
        public int T_EARTH => Data.T_EARTH;
        public int T_WIND => Data.T_WIND;
        public int T_LIGHT => Data.T_LIGHT;
        public int T_DARK => Data.T_DARK;
    }
}

File: 코드샘플/메타토이드래곤즈/코코스크리에이터/Scripts/Data/ElementData.ts
import { DataBase } from './DataBase';

/**
 * Predefined variables
 * Name = ElementData
 * DateTime = Mon Feb 21 2022 15:55:41 GMT+0900 (대한민국 표준시)
 * Author = ahnhyeon5017
 * FileBasename = ElementData.ts
 * FileBasenameNoExtension = ElementData
 * URL = db://assets/Scripts/Data/ElementData.ts
 * ManualUrl = https://docs.cocos.com/creator/3.3/manual/en/
 *
 */

export enum eElementType {
    None = 0,
    FIRE = 1,
    WATER = 2,
    EARTH = 3,
    WIND = 4,
    LIGHT = 5,
    DARK = 6
}

export class ElementRateData extends DataBase {
    protected a_element: string = "";
    public get A_ELEMENT(): string {
        return this.a_element;
    }
    public set A_ELEMENT(value: string) {
        this.a_element = value;
    }
    protected t_fire: number = -1;
    public get T_FIRE(): number {
        return this.t_fire;
    }
    public set T_FIRE(value: number) {
        this.t_fire = value;
    }
    protected t_water: number = -1;
    public get T_WATER(): number {
        return this.t_water;
    }
    public set T_WATER(value: number) {
        this.t_water = value;
    }
    protected t_earth: number = -1;
    public get T_EARTH(): number {
        return this.t_earth;
    }
    public set T_EARTH(value: number) {
        this.t_earth = value;
    }
    protected t_wind: number = -1;
    public get T_WIND(): number {
        return this.t_wind;
    }
    public set T_WIND(value: number) {
        this.t_wind = value;
    }
    protected t_light: number = -1;
    public get T_LIGHT(): number {
        return this.t_light;
    }
    public set T_LIGHT(value: number) {
        this.t_light = value;
    }
    protected t_dark: number = -1;
    public get T_DARK(): number {
        return this.t_dark;
    }
    public set T_DARK(value: number) {
        this.t_dark = value;
    }
}

/**
 * [1] Class member could be defined like this.
 * [2] Use `property` decorator if your want the member to be serializable.
 * [3] Your initialization goes here.
 * [4] Your update function goes here.
 *
 * Learn more about scripting: https://docs.cocos.com/creator/3.3/manual/en/scripting/
 * Learn more about CCClass: https://docs.cocos.com/creator/3.3/manual/en/scripting/ccclass.html
 * Learn more about life-cycle callbacks: https://docs.cocos.com/creator/3.3/manual/en/scripting/life-cycle-callbacks.html
 */
